lib/Analysis/Allocation.cpp

```diff
@@ -135,10 +135,20 @@ getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
   // TODO(jlebar): This is suboptimal if repShape[in/outOrd[0]] is small.  We
   // might be able to merge the few most-minor dimensions and get a larger
   // vector.
-  // For conversions to MmaV1 (Nvidia V100), this inVec is hardcoded in the
-  // codegen.
   inVec = std::min(srcContigPerThread[inOrd[0]], dstContigPerThread[inOrd[0]]);
   outVec = dstContigPerThread[outOrd[0]];
+
+  if (srcLayout.isa<BlockedEncodingAttr>() &&
+      dstLayout.isa<BlockedEncodingAttr>() && outOrd[0] != (rank - 1) &&
+      inOrd[0] != outOrd[0]) {
```

Comment:
Still not sure this is quite enough; what about `1x4x8x1 -> 1x8x4x1`?

I think you want to ensure that there are "enough" elements in the combined minor dims that are not transposed?

Reply:
I think you are right but I'm a little concerned about doing premature optimizations here as the heuristic would be fairly complicated to catch all the cases.

Reply:
Sounds fine to me; could we leave a TODO indicating that this may be suboptimal?

(Sorry to nitpick, but the code before I modified it had cases like "get rid of this when it's no longer necessary" and I was like, um...  I think I may have gotten rid of something that was still necessary, actually, and you may be cleaning up my mess.  :)

Reply:
sure added a TODO. Yes before you changed the code there was a TODO about removing this logic but it turns out this was the right thing to do for perf...
Note that this is still not great for perf. Phil and I discussed it and there are more opportunity to make those more efficient by using shuffle that would allow us to use wider load/store.

Reply:
> Note that this is still not great for perf. Phil and I discussed it and there are more opportunity to make those more efficient by using shuffle that would allow us to use wider load/store.

Yeah, I was thinking about this... I guess a shuffle should be a lot less expensive than a bank conflict? May need CuTe layout for the shmem to handle the swizzling?  XD

Reply:
Thanks for cleaning up my mess, Thomas.